### ISSUE TITLE
Assertion failure in TypeCheckerImpl::visitBinaryConstraint

### ISSUE THREAD

**Issue by langston-barrett** [souffle-lang/souffle#1749]
On commit 07c6db0b229f465bebd007c9095fea0c8d291bcb (and on version 2.0.2), Souffle exit with an assertion failure like this:
```
./src/souffle crash.dl
souffle: ast/transform/TypeChecker.cpp:555: virtual void souffle::ast::transform::TypeCheckerImpl::visitBinaryConstraint(const souffle::ast::BinaryConstraint&): Assertion `(leftTypes.size() == 1) && (rightTypes.size() == 1)' failed.
Aborted (core dumped)
```
when run on this program:
```
ShortestPath(x, y, a) :- Length(x, y, _), a = min Z : Length(x, y, Z).
```

**Comment by azreika** [comment]
Ha - interesting case where the missing relation declaration means we can't ground some type information. Declaring either relation (`ShortestPath` or `Length`) gives us enough information. Instead of an assertion failure, that type check should be skipped at that stage, and handled by the deduction-failure stage later on, as with all invalid typesets.

**Comment by azreika** [comment]
Fixed by PR #1750.
